packages/core/src/interpreter.ts

```diff
@@ -500,19 +498,19 @@ export class Actor<TLogic extends AnyActorLogic>
   }
 
   // TODO: make private (and figure out a way to do this within the machine)
-  public delaySend({
-    event,
-    id,
-    delay,
-    to
-  }: {
+  public delaySend(params: {
     event: EventObject;
     id: string | undefined;
     delay: number;
     to?: AnyActorRef;
   }): void {
+    const { event, id, delay } = params;
     const timerId = this.clock.setTimeout(() => {
-      this.system._relay(this, to ?? this, event as EventFromLogic<TLogic>);
+      this.system._relay(
+        this,
+        params.to ?? this,
```

Comment:
TODO: this likely deserves a test case

Reply:
done https://github.com/statelyai/xstate/pull/4351/commits/8973e43ba6dde285b925fc5c198c9d1727765c45